### PR TITLE
[NFC][Clang] Move functions of BranchProtectionInfo out of line

### PR DESCRIPTION
**Pull request by DanielKristofKiss**
Also let's add const to the setFnAttributes.

### CODE REVIEW

File: clang/include/clang/Basic/TargetInfo.h [hunk 1]
```diff
@@ -34,7 +34,6 @@
 #include "llvm/Frontend/OpenMP/OMPGridValues.h"
 #include "llvm/IR/Attributes.h"
```
Comment (nikic) [inline clang/include/clang/Basic/TargetInfo.h:35]: Please also drop the Attributes.h include.

**Review by nikic — APPROVED** [review]
LGTM, thanks!

**Comment by DanielKristofKiss** [comment]
in some build config `TargetInfo.cpp` is built without dependency to `LLVM-Core`.

**Comment by nikic** [comment]
> in some build config `TargetInfo.cpp` is built without dependency to `LLVM-Core`.

Okay, that means that putting this code inside Basic/ is a layering violation. You need to move it into CodeGen/. Probably best to revert the whole change in the meantime.

**Comment by DanielKristofKiss** [comment]
> > in some build config `TargetInfo.cpp` is built without dependency to `LLVM-Core`.
> 
> Okay, that means that putting this code inside Basic/ is a layering violation. You need to move it into CodeGen/. Probably best to revert the whole change in the meantime.

Already reverted, I'm on it.

**Pull request merged**